### PR TITLE
Implement function startingPhaseOfTetromino

### PR DESCRIPTION
**Pull request by PashaBeshchuk**

### CODE REVIEW

File: tetris.js [hunk 1]
```diff
@@ -518,7 +518,64 @@ module.exports = {
 		let arrayTypeOfTetramino = ["L", "J", "T", "S", "Z", "I", "O"];
 		return arrayTypeOfTetramino[Math.floor(Math.random() * arrayTypeOfTetramino.length)];
 	},
-	initCoordinates: function (typeOfTetromino, fieldSize ) {
-
+	initCoordinates: function (typeOfTetromino, fieldSize) {
```
Comment (zbstof) [inline tetris.js]: Rewrite this in graphical way

Comment (PashaBeshchuk) [reply tetris.js]: what do you mean?
I don't understand :(

File: test/test-initCoordinates.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+var assert = require('chai').assert;
+var tetris = require('../tetris');
+describe("Initial coordinates tetromino", function () {
```
Comment (zbstof) [inline test/test-initCoordinates.js]: add more tests for other sizes of field - 5x2, 6x2

File: tetris.js [hunk 3]
```diff
@@ -520,62 +520,62 @@ module.exports = {
...
-			coordinatesTetromino.y = k;
-			arrayForTetromino.push(coordinatesTetromino);
+			shift.x = shift.x - 1
+			arrayForTetromino = this.getCoordinates(`
+				XXXX
+			`)
+			newCoordinates = this.shiftCoordinates(arrayForTetromino, shift)
+		}
+		if (typeOfTetromino === "O") {
+			shift.x = shift.x
```
Comment (zbstof) [inline tetris.js]: Why?

File: tetris.js [hunk 4]
```diff
@@ -522,60 +522,52 @@ module.exports = {
 		let localFieldSize = Object.assign({}, fieldSize);
```
Comment (zbstof) [inline tetris.js]: why do you need a copy of the field? Size will not change if you make a copy

File: tetris.js [hunk 5]
```diff
@@ -495,5 +495,79 @@ module.exports = {
...
+				`);
+				break
+			case "O":
+				arrayForTetromino = this.getCoordinates(`
+					XX
+					XX
+				`);
+				break;
+		}
+		newCoordinates = this.shiftCoordinates(arrayForTetromino, shift);
```
Comment (zbstof) [inline tetris.js]: why not return immediately, why extra variable?

File: tetris.js [hunk 6]
```diff
@@ -495,5 +495,79 @@ module.exports = {
...
+				return "horizontal";
+			case "I":
+				return "vertical";
+			case "O":
+				return "stable";
+		}
+	},
+	getRandomTypeOfTetromino: function () {
+		let arrayTypeOfTetramino = ["L", "J", "T", "S", "Z", "I", "O"];
+		return arrayTypeOfTetramino[Math.floor(Math.random() * arrayTypeOfTetramino.length)];
```
Comment (zbstof) [inline tetris.js]: no changes needed, but for your information, please read this:
https://tetris.fandom.com/wiki/Random_Generator

File: tetris.js [hunk 7]
```diff
@@ -567,7 +566,7 @@ module.exports = {
 				`);
 				break;
 		}
-		newCoordinates = this.shiftCoordinates(arrayForTetromino, shift);
-		return newCoordinates
+		return newCoordinates = this.shiftCoordinates(arrayForTetromino, shift);
```
Comment (zbstof) [inline tetris.js]: `return newCoordinates = this.shiftCoordinates(arrayForTetromino, shift);`
is not correct syntax

**Pull request merged**